src/shared/services/addresses/addresses.service.spec.ts

```diff
@@ -0,0 +1,91 @@
+import {AddressesService} from './addresses.service';
+import {getTestBed, TestBed} from '@angular/core/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {AbstractAppConfig} from "../../../app.config";
+import {Observable} from "rxjs";
+import {HttpService} from "../http";
+import {Response, ResponseOptions} from "@angular/http";
+
+
+describe('AddressesService', () => {
+
+  let addressesService: AddressesService;
+  let appConfig;
+  let httpService;
+  let httpTestingController: HttpTestingController;
+  let validPostCode = 'SW1A 2AA';
+  let injector: TestBed;
+  let httpMock: HttpTestingController;
+  const data: any = require('../../fixture/valid-postcode-results.json');
+
+  beforeEach(() => {
+
+    appConfig = jasmine.createSpyObj<AbstractAppConfig>('AppConfig', ['getPostcodeLookupUrl',
+      'getPostcodeLookupApiKey']);
+    httpService = jasmine.createSpyObj<HttpService>('HttpService', ['get']);
+    appConfig.getPostcodeLookupUrl.and.returnValue('http://postcodeUrl/postcode=${postcode}&key=${key}');
+    appConfig.getPostcodeLookupApiKey.and.returnValue('apikey');
+    let postCodeResponse = new Response(<ResponseOptions>{body: JSON.stringify(data)});
+    httpService.get.and.returnValue(Observable.of(postCodeResponse));
+    // httpService.get.and.returnValue(Observable.of([]));
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [AddressesService, {provide: AbstractAppConfig, useValue: appConfig}, {
+        provide: HttpService,
+        useValue: httpService
+      }]
+    });
+    injector = getTestBed();
+    addressesService = injector.get(AddressesService);
+    httpMock = injector.get(HttpTestingController);
+    httpTestingController = TestBed.get(HttpTestingController);
+  });
+
+  it('should be created with all the dependencies', () => {
+    expect(addressesService).toBeTruthy();
+  });
+
+  it('should return atleast more than one addresses from a given postcode', () => {
```

Comment:
What is the purpose of this test if we provide the `should return all addresses from a given postcode location`? I would remove it and go with the latter one.

Reply:
I am expecting all the addresses are parsed from the response however, to make it clear I have renamed the tests.